metis-normalization/src/main/java/eu/europeana/normalization/dates/edtf/InstantEdtfDate.java

```diff
@@ -1,168 +1,263 @@
+
+    final String prefix = year.getValue() < 0 ? "-" : "";
+    final int trailingZeros = Integer.numberOfTrailingZeros(yearPrecision.getDuration());
+    final String yearAdjusted = paddedYear.substring(0, ISO_8601_MINIMUM_YEAR_DIGITS - trailingZeros) + "X".repeat(trailingZeros);
+    return prefix + yearAdjusted;
+  }
+
   @Override
   public String toString() {
-    StringBuilder stringBuilder = new StringBuilder();
-    if (this.getEdtfDatePart() != null) {
-      stringBuilder.append(edtfDatePart.toString());
+    final StringBuilder stringBuilder = new StringBuilder();
+    if (dateBoundaryType != DECLARED) {
+      stringBuilder.append(dateBoundaryType.getSerializedRepresentation());
+    } else if (abs(year.getValue()) > THRESHOLD_4_DIGITS_YEAR) {
+      stringBuilder.append(InstantEdtfDateBuilder.OVER_4_DIGITS_YEAR_PREFIX).append(year.getValue());
+    } else {
+      stringBuilder.append(serializeYear());
+
+      final DecimalFormat decimalFormat = new DecimalFormat("00");
+      stringBuilder.append(
+          ofNullable(month).map(Month::getValue).map(decimalFormat::format).map(m -> "-" + m).orElse(""));
+      stringBuilder.append(
+          ofNullable(yearMonthDay).map(LocalDate::getDayOfMonth).map(decimalFormat::format).map(d -> "-" + d).orElse(""));
     }
+    stringBuilder.append(dateQualification.getCharacter());
     return stringBuilder.toString();
   }
+
+  @Override
+  public int compareTo(InstantEdtfDate other) {
+    int comparatorValue = this.year.compareTo(other.year);
+    if (comparatorValue == 0 && this.month != null && other.month != null) {
+      comparatorValue = this.month.compareTo(other.month);
+      if (comparatorValue == 0 && this.yearMonthDay != null && other.yearMonthDay != null) {
+        comparatorValue = this.yearMonthDay.compareTo(other.yearMonthDay);
+      }
+    }
+    return comparatorValue;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    InstantEdtfDate that = (InstantEdtfDate) o;
+    return yearPrecision == that.yearPrecision && Objects.equals(year, that.year) && Objects.equals(month,
+        that.month) && Objects.equals(yearMonthDay, that.yearMonthDay) && dateQualification == that.dateQualification
+        && dateBoundaryType == that.dateBoundaryType;
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(yearPrecision, year, month, yearMonthDay, dateQualification, dateBoundaryType);
+  }
+
+  public Year getYear() {
+    return year;
+  }
+
+  public Month getMonth() {
+    return month;
+  }
+
+  public LocalDate getYearMonthDay() {
+    return yearMonthDay;
+  }
+
+  public YearPrecision getYearPrecision() {
+    return yearPrecision;
+  }
+
+  public DateQualification getDateQualification() {
+    return dateQualification;
+  }
+
+  public DateBoundaryType getDateEdgeType() {
```

Comment:
Rename of the method `getDateEdgeType()` => `getDateBoundaryType()`

Reply:
Done.